lib/money/money.rb

```diff
@@ -189,6 +189,52 @@ def allocate(splits)
     amounts.collect { |cents| Money.from_cents(cents) }
   end
 
+  # Allocates money between different parties up to the maximum amounts specified.
+  # Left over pennies will be assigned round-robin up to the maximum specified.
+  # Pennies are dropped when the maxima are attained.
+  #
+  # @example
+  #   Money.new(30.75).allocate_amounts([Money.new(26), Money.new(4.75)])
+  #     #=> [Money.new(26), Money.new(4.75)]
+  #
+  #   Money.new(30.75).allocate_amounts([Money.new(26), Money.new(4.74)]
+  #     #=> [Money.new(26), Money.new(4.74)]
+  #
+  #   Money.new(30).allocate_amounts([Money.new(15), Money.new(15)]
+  #     #=> [Money.new(15), Money.new(15)]
+  #
+  #   Money.new(1).allocate_amounts([Money.new(33), Money.new(33), Money.new(33)])
+  #     #=> [Money.new(0.34), Money.new(0.33), Money.new(0.33)]
+  #
+  #   Money.new(100).allocate_amounts([Money.new(5), Money.new(2)])
+  #     #=> [Money.new(5), Money.new(2)]
+  def allocate_amounts(maxima)
+    cents_maxima = maxima.map { |maximum| maximum.to_money.cents }
+    cents_total = cents_maxima.sum
+    splits = cents_maxima.map do |cents_maximum_amount|
+      BigDecimal.new(cents_maximum_amount.to_s) / cents_total
+    end
+
+    cents_amounts, left_over = amounts_from_splits(1, splits, [cents, cents_total].min)
+    if left_over.zero?
+      amounts = cents_amounts.collect { |cents| Money.from_cents(cents) }
+      return amounts
+    end
+
+    while(left_over > 0) do
+      cents_amounts.zip(cents_maxima).each.with_index do |(amount, maximum), index|
+        if amount < maximum
+          cents_amounts[index] += 1
+          left_over -= 1
+        end
+        break if left_over <= 0
+      end
+      break if cents_amounts.zip(cents_maxima).none? { |(amount, maximum)| amount < maximum }
+    end
+
+    cents_amounts.collect { |cents| Money.from_cents(cents) }
```

Comment:
This is repeated from the line 220. So maybe we can simplify this code extracting the while loop to a method and only calling it if `left_over` is not zero.